marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/getReferences.sjs

```diff
@@ -0,0 +1,33 @@
+/**
+ Copyright (c) 2021 MarkLogic Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-mapping", "execute");
+
+const core = require('/data-hub/5/artifacts/core.sjs')
+
+var stepName;
+const mappingStep = core.getArtifact("mapping", stepName);
+let resp = [{"name": "$URI", "description": "The uri of the source document"}];
+const modulePath = mappingStep.mappingParametersModulePath;
+if (modulePath) {
+  try {
+    const paramsFunction = require(modulePath)["getParameterDefinitions"];
+    const userParams = paramsFunction(mappingStep);
+    userParams.map(userParam => userParam.name = "$" + userParam.name);
```

Comment:
The use of "map" is a little confusing here because "map" is typically used to produce an output array. I think "forEach" is better here, which makes it clear we're modifying the array and not making a new one.